alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertIndices.kt

```diff
@@ -235,12 +260,61 @@ class AlertIndices(
                 .mapping(MAPPING_TYPE, alertMapping(), XContentType.JSON)
         request.addMaxIndexDocsCondition(historyMaxDocs)
         request.addMaxIndexAgeCondition(historyMaxAge)
-        val response = client.rolloversIndex(request).actionGet(requestTimeout)
+        val response = client.admin().indices().rolloversIndex(request).actionGet(requestTimeout)
         if (!response.isRolledOver) {
             logger.info("$HISTORY_WRITE_INDEX not rolled over. Conditions were: ${response.conditionStatus}")
         } else {
             lastRolloverTime = TimeValue.timeValueMillis(threadPool.absoluteTimeInMillis())
         }
         return response.isRolledOver
     }
+
+    private fun deleteOldHistoryIndices() {
+        val indicesToDelete = mutableListOf<String>()
+
+        val clusterStateRequest = ClusterStateRequest()
+                .clear()
+                .indices(HISTORY_ALL)
+                .metaData(true)
+                .local(true)
+                .indicesOptions(IndicesOptions.strictExpand())
+
+        val clusterStateResponse = client.admin().cluster().state(clusterStateRequest).actionGet()
+
+        for (entry in clusterStateResponse.state.metaData.indices) {
+            val indexMetaData = entry.value
+            val creationTime = indexMetaData.creationDate
+
+            if ((Instant.now().toEpochMilli() - creationTime) > historyRetentionPeriod.millis) {
+                val alias = indexMetaData.aliases.firstOrNull { HISTORY_WRITE_INDEX == it.value.alias }
+                if (alias != null && historyEnabled) {
```

Comment:
If `historyEnabled = false`, it means we'll end up deleting the history index w/ the `HISTORY_WRITE_INDEX` alias. But, the `historyIndexInitialized` could still be cached as true and if the user re-enables history then a new history index w/ the `HISTORY_WRITE_INDEX` will not be created and things will start to fail.

Reply:
Good catch, added logic to reset ```historyIndexInitialized``` in this case to allow the history index to be recreated when ```historyEnabled``` is set back to ```true```